go/datas/http_batch_store.go

```diff
@@ -448,8 +470,7 @@ func (bhcs *httpBatchStore) Root() hash.Hash {
 // UpdateRoot flushes outstanding writes to the backing ChunkStore before updating its Root, because it's almost certainly the case that the caller wants to point that root at some recently-Put Chunk.
 func (bhcs *httpBatchStore) UpdateRoot(current, last hash.Hash) bool {
 	// POST http://<host>/root?current=<ref>&last=<ref>. Response will be 200 on success, 409 if current is outdated.
-	bhcs.Flush()
-
+	bhcs.flushInternal(nbs.InsertOrder)
```

Comment:
This is the thing I was working on changing after I sent this patch, because I also don't like that this relies on making `noms sync` only thing that calls BatchStore.Flush(). This patch made that so, but there's no guarantee at all that it would remain that way.

Something mysteriously broke when I changed things that way, so I'll need to chase that down